crates/typst-syntax/build.rs

```diff
@@ -0,0 +1,7 @@
+fn main() {
+    println!("cargo:rerun-if-env-changed=TYPST_VERSION");
+
+    if option_env!("TYPST_VERSION").is_none() {
+        println!("cargo:rustc-env=TYPST_VERSION={}", env!("CARGO_PKG_VERSION"));
+    }
```

Comment:
Why redefine `TYPST_VERSION` if `CARGO_PKG_VERSION` is fetched separately? It also might be invalid, so having an up-to-date backup is important.

```suggestion
    println!("cargo:rerun-if-env-changed=TYPST_VERSION");
    println!("cargo:rerun-if-env-changed=CARGO_PKG_VERSION");
```

Reply:
I don't know about *backup*. If I were a packager and a project gave me an env variable (`TYPST_VERSION` in this case) to "override" what the package reports as its version, I'd like this mechanism to "work" or at least break properly.

The way I understand your proposal, you try to parse `TYPST_VERSION` and fall back to `CARGO_PKG_VERSION` if the former doesn't exist **or** doesn't parse into a valid semver object. As a package maintainer, I'd expect the code to explode if my input (to `TYPST_VERSION`) is invalid, rather than silently having it overwritten.

Or did I misunderstand your code?

Reply:
Considering committed behavior, why check for `CARGO_PKG_VERSION` [here](https://github.com/typst/typst/blob/a73515551ccddafb18090b3fa1c0785a02248b7a/crates/typst-syntax/src/version.rs#L50) at all, if at compile time `TYPST_VERSION` is **already** equal to `CARGO_PKG_VERSION`? Makes no sense.

`build.rs` and `TypstVersion::new()` must correctly negotiate env vars and their values to work as expected. One depends on the other.